server/src/main/java/keywhiz/service/resources/automation/v2/BackfillRowHmacResource.java

```diff
@@ -45,6 +47,32 @@ public BackfillRowHmacResource(DSLContext jooq, RowHmacGenerator rowHmacGenerato
     this.rowHmacGenerator = rowHmacGenerator;
   }
 
+  /**
+   * Backfill row_hmac for a specific secret.
+   */
+  @Timed @ExceptionMetered
+  @Path("secret/name/{secretName}")
+  @POST
+  @Consumes(APPLICATION_JSON)
+  @Produces(APPLICATION_JSON)
+  public void backfillSecretRowHmacByName(@PathParam("secretName") String secretName) {
+    Optional<SecretsRecord> maybeRow = jooq.selectFrom(SECRETS)
+        .where(SECRETS.NAME.eq(secretName))
+        .fetchOptional();
+
+    if (!maybeRow.isPresent()) {
+      throw new NotFoundException();
+    }
+
+    SecretsRecord row = maybeRow.get();
```

Comment:
Should we return the new hmac or some information that the HMAC has changed?

Reply:
AFAICT the row-level HMACs are an internal implementation detail and are never exposed via any of the existing APIs/models, so I didn't want to be the first to take that step. Would logging work? I don't see any obvious HTTP response codes that we could use to distinguish changed/not changed.